File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
         int ans=-1;
            for(int i=0;i<nums.size();i++)
            {
                    if(nums[i]>=target)
                    {
                            ans= i;
                            break;
                    }
            }
            if(ans == -1) return nums.size();
            return ans;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int s =0;
        int e = nums.length-1;
        if(target < nums[0]) return 0;
        if(target > nums[e]) return e+1;
        
        int ans = -1;
        
        while(s<=e)
        {
            int mid = s + (e-s)/2;
            if(nums[mid] >= target) {
                ans  = mid;
                e = mid-1;
            }
            else s = mid+1;
        }
        
        return ans;
    }
}